src/modules/cmdpal/Microsoft.CmdPal.UI/RunHistoryService.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CmdPal.Common.Services;
+using Microsoft.CmdPal.UI.ViewModels;
+
+namespace Microsoft.CmdPal.UI;
+
+internal sealed class RunHistoryService : IRunHistoryService
+{
+    private readonly AppStateModel _appStateModel;
+
+    public RunHistoryService(AppStateModel appStateModel)
+    {
+        _appStateModel = appStateModel;
+    }
+
+    public IReadOnlyList<string> GetRunHistory()
+    {
+        if (_appStateModel.RunHistory.Count == 0)
+        {
+            var history = Microsoft.Terminal.UI.RunHistory.CreateRunHistory();
+            _appStateModel.RunHistory.AddRange(history);
+        }
+
+        return _appStateModel.RunHistory;
+    }
+
+    public void ClearRunHistory()
+    {
+        _appStateModel.RunHistory.Clear();
+    }
+
+    public void AddRunHistoryItem(string item)
+    {
+        // insert at the beginning of the list
+        if (string.IsNullOrWhiteSpace(item))
+        {
+            return; // Do not add empty or whitespace items
+        }
+
+        _appStateModel.RunHistory.Remove(item);
+
+        // Add the item to the front of the history
+        _appStateModel.RunHistory.Insert(0, item);
```

Comment:
I see you're inserting this at position zero. So, the latest command should be at the top. However, when it's running, it's backwards. If I stop and restart CmdPal, the order is correct.

https://github.com/user-attachments/assets/b2f317e4-081e-49b1-80ec-bf0c599b1159